tap_hubspot/__init__.py

```diff
@@ -1047,39 +1046,48 @@ def sync_workflows(STATE, ctx):
     return STATE
 
 def sync_owners(STATE, ctx):
+    """
+    Function to sync `owners` stream records
+    """
     catalog = ctx.get_catalog_from_id(singer.get_currently_syncing(STATE))
     mdata = metadata.to_map(catalog.get('metadata'))
-    schema = load_schema("owners")
-    bookmark_key = 'updatedAt'
+    stream_id = "owners"
+    primary_key = "id"
+    bookmark_key = "updatedAt"
 
-    singer.write_schema("owners", schema, ["ownerId"], [bookmark_key], catalog.get('stream_alias'))
-    start = get_start(STATE, "owners", bookmark_key)
-    max_bk_value = start
+    max_bk_value = bookmark_value = utils.strptime_with_tz(
+        get_start(STATE, stream_id, bookmark_key))
+    LOGGER.info(f"sync {stream_id} from %s", bookmark_value)
 
-    LOGGER.info("sync_owners from %s", start)
+    params = {'limit': 500}
 
-    params = {}
-    if CONFIG.get('include_inactives'):
-        params['includeInactives'] = "true"
-    data = request(get_url("owners"), params).json()
+    schema = load_schema(stream_id)
+    singer.write_schema(stream_id, schema, [primary_key],
+                        [bookmark_key], catalog.get('stream_alias'))
 
-    time_extracted = utils.now()
+    url = get_url(stream_id)
 
-    with Transformer(UNIX_MILLISECONDS_INTEGER_DATETIME_PARSING) as bumble_bee:
+    with Transformer(UNIX_MILLISECONDS_INTEGER_DATETIME_PARSING) as transformer:
         # To handle records updated between start of the table sync and the end,
         # store the current sync start in the state and not move the bookmark past this value.
         sync_start_time = utils.now()
-        for row in data:
-            record = bumble_bee.transform(lift_properties_and_versions(row), schema, mdata)
-            if record[bookmark_key] >= max_bk_value:
-                max_bk_value = record[bookmark_key]
+        for row in gen_request_tickets(stream_id, url, params, 'results', "paging"):
+            # parsing the string formatted date to datetime object
+            modified_time = utils.strptime_to_utc(row[bookmark_key])
 
-            if record[bookmark_key] >= start:
-                singer.write_record("owners", record, catalog.get('stream_alias'), time_extracted=time_extracted)
+            # Checking the bookmark value is present on the record and it
+            # is greater than or equal to defined previous bookmark value
+            if modified_time and modified_time >= bookmark_value:
+                # transforms the data and filters out the selected fields from the catalog
+                record = transformer.transform(lift_properties_and_versions(row), schema, mdata)
+                singer.write_record(stream_id, record, catalog.get(
+                    'stream_alias'), time_extracted=utils.now())
+            if modified_time and modified_time >= max_bk_value:
+                max_bk_value = modified_time
```

Comment:
```suggestion
            if modified_time and modified_time >= bookmark_value:
                # transforms the data and filters out the selected fields from the catalog
                record = transformer.transform(lift_properties_and_versions(row), schema, mdata)
                singer.write_record(stream_id, record, catalog.get(
                    'stream_alias'), time_extracted=utils.now())

                if modified_time >= max_bk_value:
                    max_bk_value = modified_time
```

Reply:
Ideally, modified_time should never be `None` as per the docs. And also `updatedAt` is the automatic field so it is never expected to be None/null.